### PR TITLE
[#1469] improvement(doc): Add doc about webUI operation

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?

Compose documentation for the webUI, detailing the procedures for CRUD operations.

### Why are the changes needed?

Fix: #1269 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/webui.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
+---
+title: 'Gravitino webUI'
+slug: /webui
+keyword: webui
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+license: 'Copyright 2023 Datastrato Pvt Ltd. This software is licensed under the Apache License version 2.'
+---
+
```
Comment (jerryshao) [inline docs/webui.md:9]: Can you please add a simple paragraph to introduce the content of this doc in a general view?

File: docs/webui.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+After the configuration change, make sure to restart the gravitino server.
+:::
+
+### Simple mode
+
+```
+gravitino.authenticator = simple
+```
+
+If the configuration parameter for `gravitino.authenticator` is set to `simple`, the webUI will directly display the homepage (Metalakes).
```
Comment (jerryshao) [inline docs/webui.md:30]: Don't use the future tense, just using the simple tense: the web UI directly displays the homepage...

File: docs/webui.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+license: 'Copyright 2023 Datastrato Pvt Ltd. This software is licensed under the Apache License version 2.'
+---
+
+> A graphical interface accessible through a web browser.
+
+Complete the installation, and open the browser [http://localhost:8090](http://localhost:8090).
+
+## Initial page
+
+The webUI homepage displaying in Gravitino depends on the configuration parameter for OAuth mode, see the details in [Security](security.md).
```
Comment (jerryshao) [inline docs/webui.md:16]: web UI, not webUI.

File: docs/webui.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+2. **Comment**(_optional_): the comment of this metalake
+3. **Properties**(_optional_): clicking on the `ADD PROPERTY` button to add custom properties
+
+![metalake-list](assets/webui/metalake-list.png)
+
+Each metalake consists of three actions.
+
+![metalake-actions](assets/webui/metalake-actions.png)
+
+#### 1. Show metalake details
```
Comment (jerryshao) [inline docs/webui.md:84]: I think you can remove this "1. ", "2. ", "3. " here and below.

File: docs/webui.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+2. **Type**(**_required_**): the default value is `relational`,
+3. **Provider**(**_required_**): `hive`/`iceberg`/`mysql`/`postgresql`
+4. **Comment**(_optional_): the comment of this catalog
+5. **Properties**(**specific provider inculdes required fileds**)
+
+##### Providers
+
+> Required properties in various providers
+
+- hive
```
Comment (jerryshao) [inline docs/webui.md:130]: Capitalize the first letter for Hive, Iceberg, etc.

File: docs/webui.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+Following: [Apache Hive catalog](apache-hive-catalog)
```
Comment (jerryshao) [inline docs/webui.md:132]: Follow the Hive catalog doc...

File: docs/webui.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+After vertified inputing these fields, clicking on the `CREATE` button to creates a catalog.
+
+![created-catalog](assets/webui/created-catalog.png)
+
+Clicking the icon button - `←(left arrow)` will redirect to metalake page.
+
+### Schema
+
+Under Construction...
```
Comment (jerryshao) [inline docs/webui.md:202]: You'd better have a basic list view and detailed view for schema and table.

File: docs/webui.md (hunk 2 above)
Comment (ch3yne) [reply docs/webui.md]: > Don't use the future tense, just using the simple tense: the web UI directly displays the homepage...

DONE

File: docs/webui.md (hunk 3 above)
Comment (ch3yne) [reply docs/webui.md]: > web UI, not webUI.

DONE

File: docs/webui.md (hunk 4 above)
Comment (ch3yne) [reply docs/webui.md]: > I think you can remove this "1. ", "2. ", "3. " here and below.

DONE

File: docs/webui.md (hunk 5 above)
Comment (ch3yne) [reply docs/webui.md]: > Capitalize the first letter for Hive, Iceberg, etc.

DONE

File: docs/webui.md (hunk 6 above)
Comment (ch3yne) [reply docs/webui.md]: > Follow the Hive catalog doc...

DONE

File: docs/webui.md (hunk 7 above)
Comment (ch3yne) [reply docs/webui.md:218]: > You'd better have a basic list view and detailed view for schema and table.

Is the following format acceptable?
image

File: docs/webui.md [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+
+This document primarily outlines how developers can manage metadata within Gravitino using web UI, the graphical interface accessible through a web browser as an alternative to manually writing code.
+
+Currently, you can integrate [OAuth settings](security.md) to view, add, modify, and delete metalakes, create catalogs, and view catalogs, schemas, and tables, among other functions.
+
+:::caution
+More features are under development. For detailed capabilities, please refer to the [Manage metadata using Gravitino](manage-metadata-using-gravitino.md) document.
+:::
+
+Complete the installation, and open the browser [http://localhost:8090](http://localhost:8090).
```
Comment (justinmclean) [inline docs/webui.md:21]: Link to document on how to "Completer the installation". It is probably better to say "Build and deploy the Gravitino Web UI and open it in a browser at ..."

File: docs/webui.md [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+After the configuration change, make sure to restart the gravitino server.
+:::
+
+### Simple mode
+
+```
+gravitino.authenticator = simple
+```
+
+If the configuration parameter for `gravitino.authenticator` is set to `simple`, the web UI directly displays the homepage (Metalakes).
```
Comment (justinmclean) [inline docs/webui.md:39]: Use active voice

File: docs/webui.md [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+![webui-metalakes-simple](assets/webui/metalakes-simple.png)
+
+At the top-right, the current version is displayed.
```
Comment (justinmclean) [inline docs/webui.md:43]: use active voice

File: docs/webui.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+
+The main content displays a created metalake list.
+
+### Oauth mode
+
+```
+gravitino.authenticator = oauth
+```
+
+If the configuration parameter for `gravitino.authenticator` is set to `oauth`, the web UI will directly display the login page.
```
[the remark this answers is not in the document]
Comment (justinmclean) [reply docs/webui.md:53]: use active voice

File: docs/webui.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+## Manage metadata
+
+> All the manage actions are from [REST API](api/rest/gravitino-rest-api)
+
+### Metalake
+
+#### [Create a metalake](getting-started#using-rest-to-interact-with-gravitino)
+
+On the homepage, clicking the `CREATE METALAKE` button will display a dialog to create a metalake.
```
Comment (justinmclean) [inline docs/webui.md:77]: button displays

File: docs/webui.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+
+#### Edit metalake
+
+Displays a dialog for modifying fields.
+
+![create-metalake-dialog](assets/webui/create-metalake-dialog.png)
+
+#### Delete metalake
+
+Display a confirmation dialog, clicking `SUBMIT` will delete this metalake.
```
[the remark this answers is not in the document]
Comment (justinmclean) [reply docs/webui.md:105]: will delete -> deletes

**Review by justinmclean — COMMENTED** [review]
A few minor things to correct.

File: docs/webui.md [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+keyword: webui
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+license: 'Copyright 2023 Datastrato Pvt Ltd. This software is licensed under the Apache License version 2.'
+---
+import Image from '@theme/IdealImage'
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+This document primarily outlines how developers can manage metadata within Gravitino using web UI, the graphical interface accessible through a web browser as an alternative to manually writing code.
```
[the remark this answers is not in the document]
Comment (ch3yne) [reply docs/webui.md]: > users instead of developers?

DONE

[the remark this answers is not in the document]
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+
+#### [Create a metalake](getting-started#using-rest-to-interact-with-gravitino)
+
+On the homepage, clicking the `CREATE METALAKE` button will display a dialog to create a metalake.
+
+![create-metalake-dialog](assets/webui/create-metalake-dialog.png)
+
+Create a metalake needs these fields:
+
+1. **Name**(**_required_**): the metalake name, It's best to use only English letters, and avoid using special characters like _space_.
```
[the remark this answers is not in the document]
Comment (ch3yne) [reply docs/webui.md:84]: Currently, there should be no restrictions; however, for conventional naming, the use of special characters should be avoided as much as possible. I will now remove this section.

File: docs/webui.md [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+2. **Comment**(_optional_): the comment of this metalake
+3. **Properties**(_optional_): clicking on the `ADD PROPERTY` button to add custom properties
+
+![metalake-list](assets/webui/metalake-list.png)
+
+Each metalake consists of three actions.
```
[the remark this answers is not in the document]
Comment (ch3yne) [reply docs/webui.md:90]: Yes, thanks!

File: docs/webui.md [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+
+If the configuration parameter for `gravitino.authenticator` is set to `oauth`, the web UI will directly display the login page.
+
+![webui-login-with-oauth](assets/webui/login-with-oauth.png)
+
+1. Follow the [example steps](security#example), enter values in these text fields.
+
+- Client ID: `test`
+- Client Secret: `test`
+- Scope: `test`
```
[the remark this answers is not in the document]
Comment (ch3yne) [reply docs/webui.md]: I have revised this description; is it now acceptable?

**Comment by ch3yne** [comment]
> A few minor things to correct.

I have revised the entire content according to the suggestions. Please check it again. Thank you very much!

File: docs/webui.md [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
+`<path-to-gravitino>/bin/gravitino.sh restart`
+:::
+
+### Simple mode
+
+```
+gravitino.authenticator = simple
+```
+
+Set the configuration parameter for `gravitino.authenticator` to `simple`, the web UI directly displays the homepage (Metalakes).
```
Comment (justinmclean) [inline docs/webui.md:42]: no need for "directly"

File: docs/webui.md [hunk 19]
```diff
@@ -0,0 +1,240 @@
...
+
+![webui-metalakes-simple](assets/webui/metalakes-simple.png)
+
+At the top-right, displays the current version.
```
Comment (justinmclean) [inline docs/webui.md:46]: At the top-right, the UI displays the current Gravitino version.

File: docs/webui.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (justinmclean) [reply docs/webui.md]: I'm guessing a full stop may cause some issues due to how we handle names.

File: docs/webui.md [hunk 20]
```diff
@@ -0,0 +1,240 @@
...
+
+Displays a confirmation dialog, clicking on the `SUBMIT` button deletes this metalake.
+
+![delete-metalake](assets/webui/delete-metalake.png)
+
+### Catalog
+
+Clicking on a metalake name in the table views catalogs in a metalake.
+
+If this is the first time, it shows no data until after creating catalog.
```
Comment (justinmclean) [inline docs/webui.md:112]: creating a catalog.

File: docs/webui.md [hunk 21]
```diff
@@ -0,0 +1,240 @@
...
+
+This document primarily outlines how users can manage metadata within Gravitino using the web UI, the graphical interface accessible through a web browser as an alterative to writing code or using the REST interface.
+
+Currently, you can integrate [OAuth settings](security.md) to view, add, modify, and delete metalakes, create catalogs, and view catalogs, schemas, and tables, among other functions.
+
+:::caution
+More features are under development. For detailed capabilities, please refer to the [Manage metadata using Gravitino](manage-metadata-using-gravitino.md) document.
+:::
+
+[Build](how-to-build#quick-start) and [deploy](getting-started#getting-started-locally) the Gravitino Web UI and open it in a browser at [http://localhost:8090](http://localhost:8090).
```
Comment (jerryshao) [inline docs/webui.md:22]: The url is not always "http://localhost:8090", it depends on how users configure gravitino host and port, I can think you can write to 

```
open it in a browser at `http://<gravitino-host>:<gravitino-port>`, by default is [http://localhost:8090](http://localhost:8090).
```

File: docs/webui.md [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+
+```
+gravitino.authenticator = oauth
+```
+
+Set the configuration parameter for `gravitino.authenticator` to `oauth`, the web UI directly displays the login page.
+
+![webui-login-with-oauth](assets/webui/login-with-oauth.png)
+
+1. Enter the values corresponding to your specific configuration. For detailed instructions, please refer to [Security](security).
```
Comment (jerryshao) [inline docs/webui.md:60]: `[Security](security.md)`

File: docs/webui.md (hunk 7 above)
Comment (jerryshao) [reply docs/webui.md:217]: Yes, it would be better you also have screenshots to show the schemas and tables.

File: docs/webui.md (hunk 18 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 19 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 20 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 21 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 22 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+license: 'Copyright 2023 Datastrato Pvt Ltd. This software is licensed under the Apache License version 2.'
+---
+
+import Image from '@theme/IdealImage'
+import Tabs from '@theme/Tabs'
+import TabItem from '@theme/TabItem'
+
+This document primarily outlines how users can manage metadata within Gravitino using the web UI, the graphical interface accessible through a web browser as an alterative to writing code or using the REST interface.
```
Comment (jerryshao) [inline docs/webui.md:14]: "the graphical interface is accessible..."

File: docs/webui.md [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+
+Currently, you can integrate [OAuth settings](security.md) to view, add, modify, and delete metalakes, create catalogs, and view catalogs, schemas, and tables, among other functions.
+
+:::caution
+More features are under development. For detailed capabilities, please refer to the [Manage metadata using Gravitino](manage-metadata-using-gravitino.md) document.
```
Comment (jerryshao) [inline docs/webui.md:19]: "For the details, please..."

File: docs/webui.md [hunk 25]
```diff
@@ -0,0 +1,240 @@
...
+:::
+
+[Build](how-to-build#quick-start) and [deploy](getting-started#getting-started-locally) the Gravitino Web UI and open it in a browser at `http://<gravitino-host>:<gravitino-port>`, by default is [http://localhost:8090](http://localhost:8090).
```
Comment (jerryshao) [inline docs/webui.md:22]: Add ".md" suffix to all the links here and below, like "how-to-build.md#quick-start"

File: docs/webui.md [hunk 26]
```diff
@@ -0,0 +1,240 @@
...
+`<path-to-gravitino>/bin/gravitino.sh restart`
+:::
+
+### Simple mode
+
+```
+gravitino.authenticator = simple
+```
+
+Set the configuration parameter for `gravitino.authenticator` to `simple`, the web UI displays the homepage (Metalakes).
```
Comment (jerryshao) [inline docs/webui.md:42]: remove "for".

File: docs/webui.md [hunk 27]
```diff
@@ -0,0 +1,240 @@
...
+
+The main content displays the existing metalake list.
+
+### Oauth mode
+
+```
+gravitino.authenticator = oauth
+```
+
+Set the configuration parameter for `gravitino.authenticator` to `oauth`, the web UI directly displays the login page.
```
Comment (jerryshao) [inline docs/webui.md:56]: Also here remove "for". And remove "directly" here.

File: docs/webui.md (hunk 23 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 24 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 25 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 26 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

File: docs/webui.md (hunk 27 above)
Comment (ch3yne) [reply docs/webui.md]: DONE

**Review by jerryshao — APPROVED** [review]

File: docs/webui.md [hunk 28]
```diff
@@ -27,180 +39,210 @@ After the configuration change, make sure to restart the gravitino server.
...
+Creating a catalog requires these fields:
 
-1. **Catalog name**(**_required_**): the catalog name of this metalke
-2. **Type**(**_required_**): the default value is `relational`,
+1. **Catalog name**(**_required_**): the name of the catalog
+2. **Type**(**_required_**): the default value is `relational`
 3. **Provider**(**_required_**): `hive`/`iceberg`/`mysql`/`postgresql`
 4. **Comment**(_optional_): the comment of this catalog
-5. **Properties**(**specific provider inculdes required fileds**)
+5. **Properties**(**specific provider included required fields**)
```
Comment (jerryshao) [inline docs/webui.md:132]: @ch3yne what's the meaning of this sentence, I cannot follow it.

Comment (jerryshao) [reply docs/webui.md:132]: each `provider` must fill in the required property fields specifically.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**